src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -197,15 +210,16 @@ struct aie_core_info_sysfs
   get(const xrt_core::device* device, key_type key)
   {
     boost::property_tree::ptree ptarray;
-    uint32_t max_col = 0, max_row = 0;
+    aie_metadata_info aie_meta;
 
-    get_aie_row_col(device, max_row, max_col);
+    get_aie_row_col(device, aie_meta);
+    std::string aiepart = std::to_string(aie_meta.shim_row)+"_"+std::to_string(aie_meta.num_cols);
 
     /* Loop each all aie core tiles and collect core, dma, events, errors, locks status. */
-    for(int i=0;i<max_col;i++)
-      for(int j=0; j<(max_row-1);j++)
+    for(int i=0;i<aie_meta.num_cols;i++)
```

Comment:
```suggestion
    for  (int i = 0; i < aie_meta.num_cols; ++i)
```
Double nesting suggests refactor.